mcs/class/System/System.Net.Sockets/SocketAsyncResult.cs

```diff
@@ -155,7 +155,11 @@ public void Complete ()
 
 			AsyncCallback callback = AsyncCallback;
 			if (callback != null) {
-				ThreadPool.UnsafeQueueUserWorkItem (_ => callback (this), null);
+				QueueUserWorkItemCallback workItemCallback = (object state) => {
```

Comment:
`SocketAsyncResult. AsyncCallback` is readonly, so there is no chance of it changing once the SocketAsyncResult has been constructed.